uk.co.compucorp.civicrm.civihrsampledata/tests/phpunit/CRM/CiviHRSampleData/Importer/AbsencePeriodTest.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+require_once EXTENSION_ROOT_DIR . 'CRM/CiviHRSampleData/Importer/AbsencePeriod.php';
+
+/**
+ * Class CRM_CiviHRSampleData_Importer_AbsencePeriodTest
+ *
+ * @group headless
+ */
+class CRM_CiviHRSampleData_Importer_AbsencePeriodTest extends CRM_CiviHRSampleData_BaseImporterTest {
+
+  private $rows;
+
+  public function setUpHeadless() {
+    return \Civi\Test::headless()
+      ->install('org.civicrm.hrabsence')
+      ->apply();
+  }
+
+  public function setUp() {
+    $this->rows = [];
+    $this->rows[] = $this->importHeadersFixture();
+  }
+
+  public function testImport() {
+    $this->rows[] = [
+      '2016',
+      '2016 (Jan 1 to Dec 31)',
+      '2016-01-01 02:00:00',
+      '2017-01-01 01:59:59',
+    ];
+
+    $this->runImporter('CRM_CiviHRSampleData_Importer_AbsencePeriod', $this->rows);
+
+    $this->assertEquals('2016', $this->apiQuickGet('HRAbsencePeriod','name', '2016'));
```

Comment:
for 21 class , and suppose at least I am going to write at least 5 extra test cases =>  5*21 = at least 105 test case .. hell this will take the eternity :smile: ... also wouldn't you think this is not testing what the code actually do .. at the end there is a BAO methods that have this kind of tests  ( eg. wrong dates parsing)... my code is just in someway a wrapper for the API ,, I know it is not an exact wrapper and there are some mapping logic happen but at the end no insertion happen to the db without the API so I think it is safe to assume that BAO tests are sufficient for this kind of stuff , what do you think ?